parcels/compilation/codecompiler.py

```diff
@@ -94,17 +94,14 @@ def __init__(self, cppargs=None, ldargs=None, incdirs=None, libdirs=None, libs=N
         opt_flags = ['-g', '-O3']
         arch_flag = ['-m64' if calcsize("P") == 8 else '-m32']
         self._cppargs = ['-Wall', '-fPIC']
-        if len(incdirs) > 0:
-            self._cppargs += Iflags
+        self._cppargs += Iflags
         self._cppargs += opt_flags + cppargs + arch_flag
         self._ldargs = ['-shared']
-        if len(Lflags) > 0:
-            self._ldargs += Lflags
-        if len(lflags) > 0:
-            self._ldargs += lflags
+        self._ldargs += Lflags
+        self._ldargs += lflags
```

Comment:
No, I haven't tried with `mpicc`, I don't have that installed. Could you try? 

And you say below that it could be made less verbose. Do you want to try that too, @CKehl?